src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -324,7 +317,28 @@ public boolean createSharedLib() throws IOException, InterruptedException {
      */
     @Override
     public boolean createStaticLib() throws IOException, InterruptedException {
-        return true;
+        if (!compile()) {
+            Logger.logSevere("Error building a static library: error compiling the native image");
+            return false;
+        }
+        if (!link()) {
+            Logger.logSevere("Error building a static library: error linking the native image");
+            return false;
+        }
+        ProcessRunner createStaticLibRunner = new ProcessRunner(getStaticLinker());
+        createStaticLibRunner.addArg(getStaticLinkerArgs());
+        Path dest = paths.getGvmPath().resolve("lib" + projectConfiguration.getAppName() + ".a");
+        createStaticLibRunner.addArg(dest.toString());
+        createStaticLibRunner.addArg(getProjectObjectFile().toString());
+        createStaticLibRunner.addArgs(getAdditionalObjectFiles());
```

Comment:
there should be no additional object files

Reply:
removed